mvn-resolver-transport-http3/src/main/java/com/artipie/aether/transport/http3/HttpTransporter.java

```diff
@@ -291,23 +290,27 @@ protected void implClose() {
         AuthenticationContext.close(proxyAuthContext);
     }
 
-    private ContentResponse makeRequest(HttpMethod method, TransportTask task)
-        throws ExecutionException, InterruptedException, TimeoutException, MalformedURLException {
+    private ContentResponse makeRequest(HttpMethod method, TransportTask task, Request.Content bodyContent) throws MalformedURLException {
         final String url = new URL(this.baseUri.toURL(), task.getLocation().toString()).toString();
         System.err.printf("Custom HttpTransporter.makeRequest() called! Method: %s; URL: %s%n", method.toString(), url);
-
         if (this.authInfo != null) {
             this.client.getAuthenticationStore().addAuthenticationResult(
                 new BasicAuthentication.BasicResult(this.baseUri, this.authInfo[0], this.authInfo[1])
             );
         }
-        final ContentResponse response = this.client.newRequest(url).method(method).headers(httpFields -> {
-            System.err.printf("\tCustom HEADER HttpTransporter.makeRequest() called! fields: %d; URL: %s%n", httpFields.size(), url);
-            final Object token = this.state.getUserToken();
-            if (token != null) {
-                httpFields.add(LocalState.USER_TOKEN, token.toString());
-            }
-        }).send();
+        final Request request = this.client.newRequest(url);
+        final ContentResponse response;
+        try {
+            response = request.method(method).headers(httpFields -> {
+                System.err.printf("\tCustom HEADER HttpTransporter.makeRequest() called! fields: %d; URL: %s%n", httpFields.size(), url);
+                final Object token = this.state.getUserToken();
```

Comment:
Why do we need a `LocalState` field?
Should `HttpTransporter` be thread safe?

Reply:
@dgarus , interenstingly I don't see thread safety notes even for state implementations. I'll recheck this deeper. 
Also, Maven assumes having local and global (cross-instance) state support for plugins, so I added local and global states too, in order to have session mechanism, more for near-future use.

Reply:
@dgarus , I think LocalState is still required for convenience, at least, since it also hides details on loading/storing parameters in local state and access to GlobalState, which sometimes could be not required sometimes, or GlobalState could be not available from Maven due to current maven configuration. So I think `LocalState` and `GlobalState` still useful, but need better javadoc on their purpose.

Reply:
> @dgarus , I think LocalState is still required for convenience, at least, since it also hides details on loading/storing parameters in local state and access to GlobalState, which sometimes could be not required sometimes, or GlobalState could be not available from Maven due to current maven configuration. So I think `LocalState` and `GlobalState` still useful, but need better javadoc on their purpose and some cleanup.

I don't understand why is `LocalState` needed? 
To hide details from whom?

Reply:
@dgarus , `LocalState` provides methods `get/set` for the parameters and encapsulates logic on concurrency and global state availability and access. Also, maybe `LocalState` should be called somehow differently, `Session` for example.

Reply:
If you say that it `encapsulates logic on concurrency` than it does it incorrect way, so I recommend you to remove the extra entity from the code.

Reply:
@dgarus , I think `LocalState` needs better description and also `synchronized` java block block in the methods for concurrent access. It already hides `GlobalState` related checks & access keys correctly.

Reply:
@dgarus , Another option actually is to remove both `LocalState` and `GlobalState`, since its not important for basic auth at the moment. And put this feature back later, when we get more session related parameters & logic.

Reply:
Yes, please, leave only what is in use.